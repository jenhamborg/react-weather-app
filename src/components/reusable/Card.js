import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const CardContainer = styled.div`
  align-content: center;
  background-color: rgba(255, 255, 255, 0.95);
  border-radius: 4px;
  color: #242526;
  text-align: center;
  height: 265px;
  width: 300px;
  h3 {
    font-size: 2rem;
  }
  .card-center {
    font-size: 4rem;
    font-weight: 700;
  }
  .card-icon-container {
    background-color: rgba(12, 46, 75, 0.25);
    margin-left: auto;
    margin-right: auto;
    height: 100px;
    width: 100px;
    border-radius: 4px;
  }
`;

export default function Card({ centerData, icon, iconDescription, title }) {
  return (
    <CardContainer>
      <h3>{title}</h3>
      <div>
        <div className="card-icon-container">
          <img
            src={`http://openweathermap.org/img/wn/${icon}@2x.png`}
            alt={iconDescription}
          />
        </div>
        <div className="card-center">{centerData}</div>
      </div>
    </CardContainer>
  );
}

Card.propTypes = {
  centerData: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  icon: PropTypes.string,
  iconDescription: PropTypes.string,
  title: PropTypes.string,
};
