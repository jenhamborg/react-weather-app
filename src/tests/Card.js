import React from "react";
import { render, cleanup } from "@testing-library/react";
import Card from "../components/reusable/card";

describe("Card", () => {
  afterEach(() => {
    cleanup;
  });

  it("Correct row data is displayed", async () => {
    const { getByText, getByTestId, getByAltText } = render(
      <Card
        centerData={"center data"}
        icon={"icon-content"}
        iconDescription={"icon description"}
        title={"title content"}
      />
    );

    const centerData = getByText("center data");
    const icon = getByTestId("card-icon");
    const iconDescription = getByAltText("icon description");
    const title = getByText("title content");

    expect(centerData).toHaveTextContent("center data");
    expect(icon).toHaveProperty(
      "src",
      "http://openweathermap.org/img/wn/icon-content@2x.png"
    );
    expect(iconDescription).toHaveProperty("alt", "icon description");
    expect(title).toHaveTextContent("title content");
  });
});
