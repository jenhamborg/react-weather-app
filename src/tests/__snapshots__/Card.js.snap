// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card__CardContainer-sc-1i2k6yt-0 jjzxfP"
>
  <h2 />
  <div>
    <div
      className="card-icon-container"
    >
      <img
        data-testid="card-icon"
        src="http://openweathermap.org/img/wn/undefined@2x.png"
      />
    </div>
    <div
      className="card-center"
    />
  </div>
</div>
`;
